slick/src/main/scala/slick/dbio/DBIOAction.scala

```diff
@@ -554,7 +561,7 @@ object SynchronousDatabaseAction {
     * fused action can fail with a `ClassCastException` during evaluation. */
   private[slick] def fuseUnsafe[R, S <: NoStream, E <: Effect](a: DBIOAction[R, S, E]): DBIOAction[R, S, E] = {
     a match {
-      case FlatMapAction(base: SynchronousDatabaseAction[_, _, _, _], f, ec) if ec eq DBIO.sameThreadExecutionContext =>
+      case FlatMapAction(base: SynchronousDatabaseAction[_, _, _, _], Vector(f), ec) if ec eq DBIO.sameThreadExecutionContext =>
```

Comment:
I haven't checked all use cases but this looks suspicious. It won't be able to fuse a `FlatMapAction` that contains multiple mapping functions. Is this good enough? Should it split off the first mapping and only fuse that? Or assume that everything can be fused?